odxtools/multiplexer.py

```diff
@@ -236,3 +236,33 @@ def decode_from_pdu(self, decode_state: DecodeState) -> ParameterValue:
         decode_state.origin_byte_position = orig_origin
 
         return result
+
+    @override
+    def get_static_bit_length(self) -> int | None:
+        """
+        Returns the static bit length of the multiplexer structure, if determinable.
+
+        If all cases (including the default, if present) have the same static bit length,
+        the codec length is considered static and is returned.
+        Otherwise, returns None to indicate that the size is dynamic.
+        """
+        reference_case = self.default_case if self.default_case else self.cases[0]
+        if reference_case.structure is None:
+            return None
+        case_bit_length = reference_case.structure.get_static_bit_length()
+        if case_bit_length is None:
+            return None
+        for mux_case in self.cases:
+            if mux_case.structure is None:
+                return None
+            case_size = mux_case.structure.get_static_bit_length()
+            if case_size != case_bit_length:
+                return None  # Found a case with a different or unknown size
+
+        switch_key_size = self.switch_key.dop.get_static_bit_length()
+        if switch_key_size is None:
+            return None
+
+        return max(
+            switch_key_size + self.switch_key.byte_position * 8 +
```

Comment:
Note that ODX allows to specify atrocities like

- 5 bytes case content @ byte position 0
- 1 byte switch key @ byte position 20
- 4 bytes value parameter @ byte position 6
- 4 bytes value parameter @ byte position 22

what even is the size of the multiplexer in this case? While I don't think that we should handle these cases in the code, I would like to see this fact mentioned in the docstring for the function above.

Reply:
it will be always the max of (byes required for key,  byes required for case)

bytes required = offset + size

current code handles all those "atrocities" just fine

Reply:
yeah, correct. If I think about it, this is an issue with any composite object in ODX, i.e., defining the static size as the minimum size required in a PDU is fine, one just needs to be aware that the sum of all static sizes is not necessarily the static size of the composite object. (IOW, the current code for structures is only "mostly correct", whereas "mostly" means for de-/encoding purposes...)